### PR TITLE
Provide sensible defaults for content translation of Open Social's content types

### PR DESCRIPTION
**Pull request by Kingdutch**
## Problem
There is no easy way to enable content translation for some of the content types that are present in the distribution. Manually configuring this on each Open Social installation is problematic when a set of sane defaults work fine.

Exposing Drupal core's content translation configuration to sitemanagers is not an option because it provides them with too much power and settings.

## Solution
This pull request introduces the social_content_translation module which provides settings that can turn content translation on and off for Open Social content types. This provides us with the ability to build a UI that can be exposed to sitemanagers in the future.

It achieves its purpose by enabling the content translation module and then disabling the content translation for all fields and entities that it manages. When the module is configured to enable content translation for a certain content type the configuration overrides will apply and set translatability for the correct fields and entity.

The configuration override classes that control what fields are translatable when enabled are stored in the modules that create the content types. This ensures that they're co-located with the modules that declare the fields that it configures and ensures that they can be kept up to date even if a module is distributed as a standalone Drupal module.

This solution is build with Open Social's Enterprise/SaaS architecture in mind which favours Plug 'n Play over long configuration screens. If you extend the Open Social core content types you have two options to hook into this system.

1. Overwrite the `ContentTranslationDefaultsConfigOverride::getTranslationOverrides` method to add configuration for the field that you added. This will ensure that the `social_content_translation.settings` configuration will also control your fields.

2. Don't use the social_content_translation module at all but just use Drupal core's content translation configuration for the fine grained control that you require.

If you're providing your own content type to Open Social then you can also choose between the above two options but may want to implement your own `contentTranslationDefaultsConfigOverride` class. A follow-up issue will have to be created to pick up services tagged with `social_language_defaults` (as in this PR). The `getModule` method can be used as configuration key in `social_content_translation.settings` and the `getDisplayName` method can be used in a UI for sitemanagers.

## Open Questions
- [ ] Is `ContentTranslationDefaultsConfigOverride` the right name or do we have something better?
- [ ] Check whether other content types also need a `ContentTranslationDefaultsConfigOverride` class

## Issue tracker
This PR has no Drupal.org issue as it replaces the following longstanding pull request: #832 

## HTT
- [ ] Check out the code changes
- [ ] Add a language to the platform besides English
- [ ] Login as sitemanager and create an event, page, book page and landing page
- [ ] See that they are not translatable
- [ ] Enable the social_content_translation module
- [ ] See that all four content types are now translatable
- [ ] Use `drush ev "\Drupal::configFactory()->getEditable('social_content_translation.settings')->set('social_event', false)->save();"` to disable translation for events (or another content type)
- [ ] See that this content type is no longer translatable (if a cache clear is required then we need to fix out cacheability of the overrides).

## Release notes
Open Social now includes sensible defaults to enable content translation for the content types that are included. To enable content translation, simply enable the social_content_translation module. 

A UI to expose these settings to sitemanagers will be added in a later release.

## Follow-ups
- [ ] UI Story (see above)
- [ ] A good place to document this for Developers that want to extend it.

### CODE REVIEW

**Comment by Kingdutch** [comment]
This seems to fix https://www.drupal.org/project/social/issues/2912385 too

**Comment by Kingdutch** [comment]
@ronaldtebrake @bramtenhove The `change-language` test is failing because the first `.navbar-nav .dropdown` element is now the language switcher instead of the `+` (add content) menu.

I'm not sure what to change to fix this.

**Comment by bramtenhove** [comment]
I also noticed a coding standard issue @Kingdutch, Travis is failing on that.

**Comment by bramtenhove** [comment]
@Kingdutch, reassigned to you. The feature state is not clean and PAC also needs work.

**Comment by ronaldtebrake** [comment]
^ i hope you can get it green on Monday, i will try to do my very best on tuesday early mornings to review it then.

File: modules/custom/social_language/src/SocialLanguageMetadataBubblingUrlGenerator.php [hunk 1]
```diff
@@ -46,7 +46,23 @@ public function generateFromRoute($name, $parameters = [], $options = [], $colle
...
 
       if ($options['language']->getId() != $language->getId()) {
-        $options['language'] = $language;
+        $reset_language = TRUE;
+        $unmodified_pages = [
+          'content_translation_overview',
+        ];
+        $current_route = \Drupal::routeMatch()->getRouteName();
+        \Drupal::moduleHandler()
+          ->alter('social_language_unmodified_pages', $unmodified_pages);
```
[the remark this answers is not in the document]
Comment (Kingdutch) [reply modules/custom/social_language/src/SocialLanguageMetadataBubblingUrlGenerator.php]: @ronaldtebrake I don't think so

**Pull request merged**